src/gausskernel/storage/access/k2/session.cpp

```diff
@@ -199,12 +172,114 @@ sh::Response<>  TxnManager::CreateCollection(const std::string& collection_name,
             .writeIOPs = 0,
             .minNodes = 1   // K2 Http proxy hangs if minNodes = 0
         },
-        .retentionPeriod = sh::Duration(1h) * 90 * 24,  //TODO: get this from config or from param in
+        .retentionPeriod = cconf.getDurationMillis("retention_period", 1h*24*90),
         .heartbeatDeadline = sh::Duration(0),
         .deleted = false
     };
 
-    return CreateCollection(metadata, rangeEnds);
+    return createCollection(metadata, rangeEnds);
+}
+
+boost::future<sh::Response<sh::dto::SKVRecord>>
+TxnManager::read(sh::dto::SKVRecord& record) {
+    K2LOG_D(log::k2pg, "read: {}", record);
+    return beginTxn()
+        .then([this, &record](auto&& beginFut) mutable {
+            auto&& [beginStatus] = beginFut.get();
+            if (!beginStatus.is2xxOK()) {
+                return sh::MakeResponse<sh::dto::SKVRecord>(std::move(beginStatus), sh::dto::SKVRecord{});
+            }
+            return _txn->read(record);
+        })
+        .unwrap();
+}
+
+boost::future<sh::Response<>>
+TxnManager::write(sh::dto::SKVRecord& record, bool erase,
+                  sh::dto::ExistencePrecondition precondition) {
+    K2LOG_D(log::k2pg, "write: {}, erase: {}, precond: {}", record, erase, precondition);
+    return beginTxn()
+        .then([this, &record, erase = erase, precondition = precondition](auto&& beginFut) mutable {
```

Comment:
Same as for read, not sure if the record capture is safe

Reply:
same contract as the underlying library - you keep the reference around for the duration of call

Reply:
fixed to accept by value